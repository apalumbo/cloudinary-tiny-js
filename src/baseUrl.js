export default ({
  cloudName,
  secure = true,
  subDomain = 'res',
  hostName = `${subDomain}.cloudinary.com`,
}) => `http${secure ? 's' : ''}://${hostName}/${cloudName}/image/upload/`
