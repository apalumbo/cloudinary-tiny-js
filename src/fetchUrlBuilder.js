export default (imageUrlBuilder) => {
  const imageFetchUrl = imageUrlBuilder('fetch')
  return (remoteUrl, transform) => imageFetchUrl(encodeURIComponent(remoteUrl), transform)
}
