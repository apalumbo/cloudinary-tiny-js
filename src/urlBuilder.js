const makeBaseUrl = ({
  cloudName,
  secure = true,
  subDomain = 'res',
  hostName = `${subDomain}.cloudinary.com`,
}, resourceType) => `http${secure ? 's' : ''}://${hostName}/${cloudName}/${resourceType}/`

const compiler = (compileParameter, defaultTransform) => {
  const compile = parameters => (
    Object.keys(parameters)
      .map(param => compileParameter(param, parameters[param]))
      .join(',')
  )

  return (transform) => !transform ? '' : (
    '/' + Array.isArray(transform)
      ? transform.map(compile).join('/')
      : compile({...defaultTransform, ...transform})
  )
}

export default (options) => (resourceType, compileParameter, defaultTransform) => {
  const baseUrl = makeBaseUrl(options, resourceType)
  const compile = compiler(compileParameter, defaultTransform)

  return type => (publicId, transform) => (
    `${baseUrl}/${type}${compile(transform)}/${publicId}`
  )
}
