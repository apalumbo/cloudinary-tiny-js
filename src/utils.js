export const invariant = (
  condition, parameter, value, message,
  source = `/image_transformation_reference#${parameter}_parameter`
) => {
  if (!condition) {
    value = typeof value === 'string' ? `'${value}'` : value
    source = `http://cloudinary.com/documentation${source}`
    throw new Error(
      `Cloudinary Image :: ${parameter} ${message}, received: ${value} - see ${source}`
    )
  }
}

export const isNumber = (value) => typeof value === 'number' || !Number.isNaN(Number(value))

export const startsWith = (needle, haystack) => haystack.indexOf(needle) === 0

export const includes = (needle, haystack) => haystack.indexOf(needle) > -1

export const shouldBeOneOf = (possibleValues) => `should be one of ['${possibleValues.join(`', '`)}']`

