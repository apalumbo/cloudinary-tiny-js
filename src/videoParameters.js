const validate = {}

const urlParameters = {}

export default function compileVideoParameter(parameter, value) {
  switch (parameter) {
    // no default
  }

  if (process.env.NODE_ENV !== 'production') {
    if (!validate[parameter]) {
      throw new Error(`Cloudinary Video :: unknown transform parameter provided: '${parameter}'`)
    }
    validate[parameter](value)
  }

  return urlParameters[parameter] + value
}
