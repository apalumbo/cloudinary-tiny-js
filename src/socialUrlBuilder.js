import { invariantSocial as invariant, includes, shouldBeOneOf } from './utils'

const supportedNetworks = ['facebook', 'twitter', 'twitterName', 'instagram', 'instagramName', 'gplus', 'gravatar']

const validateNetwork = network => invariant(
  includes(network, supportedNetworks),
  'socialOptIn', network, shouldBeOneOf(supportedNetworks)
)

export default (imageUrlBuilder, socialOptIn = []) => (
  (socialOptIn.length ? socialOptIn : supportedNetworks)
    .reduce((acc, network) => {
      process.env.NODE_ENV !== 'production' && validateNetwork(network)
      acc[network] = imageUrlBuilder(network.replace('N', '_n')) // snake case networkName
      return acc
    }, {})
)
