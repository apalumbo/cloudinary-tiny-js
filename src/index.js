import configureUrlBuilder from './urlBuilder'
import imageParameters from './parameters/image'
import videoParameters from './parameters/video'

export default function cloudinary(options, defaultTransforms = {}) {
  const urlBuilder = configureUrlBuilder(options)
  const imageUrlBuilder = urlBuilder(imageParameters, defaultTransforms.image)
  return {
    video: urlBuilder(videoParameters, defaultTransforms.video)('upload'),
    image: imageUrlBuilder('upload'),
  }
}
