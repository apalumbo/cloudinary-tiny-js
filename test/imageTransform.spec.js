import { formatParameter, compileParameter, imageTransform } from '../src/imageTransform'

describe('Image Transform Parameter', () => {
  describe('format', () => {
    it('compiles to an empty string for no value', () => {
      expect(formatParameter()).toBe('')
    })
    it('compiles to a file extension', () => {
      expect(formatParameter('jpg')).toBe('.jpg')
    })
    it('throws on an invalid extension', () => {
      expect(() => formatParameter('bad')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('width', () => {
    it('accepts a number', () => {
      expect(compileParameter('width', 300)).toBe('w_300')
    })
    it('accepts a numeric string', () => {
      expect(compileParameter('width', '300')).toBe('w_300')
    })
    it('accepts `auto`', () => {
      expect(compileParameter('width', 'auto')).toBe('w_auto')
    })
    it('accepts a string starting with `auto:`', () => {
      expect(compileParameter('width', 'auto:50')).toBe('w_auto:50')
    })
    it('throws when invalid', () => {
      expect(() => compileParameter('width', 'bad:auto')).toThrowErrorMatchingSnapshot()
      expect(() => compileParameter('width', 'auto-bad')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('height', () => {
    it('accepts a number', () => {
      expect(compileParameter('height', 300)).toBe('h_300')
    })
    it('accepts a numeric string', () => {
      expect(compileParameter('height', '300')).toBe('h_300')
    })
    it('throws when invalid', () => {
      expect(() => compileParameter('height', 'bad')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('crop', () => {
    it('accepts valid string values', () => {
      expect(compileParameter('crop', 'scale')).toBe('c_scale')
      expect(compileParameter('crop', 'fill')).toBe('c_fill')
    })
    it('throws when invalid', () => {
      expect(() => compileParameter('crop', 'bad')).toThrowErrorMatchingSnapshot()
      expect(() => compileParameter('crop', 300)).toThrowErrorMatchingSnapshot()
    })
  })

  describe('aspectRatio', () => {
    it('accepts a number', () => {
      expect(compileParameter('aspectRatio', 1.5)).toBe('ar_1.5')
    })
    it('accepts valid string values', () => {
      expect(compileParameter('aspectRatio', '1.5')).toBe('ar_1.5')
      expect(compileParameter('aspectRatio', '16:9')).toBe('ar_16:9')
    })
    it('throws when invalid', () => {
      expect(() => compileParameter('aspectRatio', 'bad')).toThrowErrorMatchingSnapshot()
      expect(() => compileParameter('aspectRatio', '7:')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('gravity', () => {
    it('accepts valid string values', () => {
      expect(compileParameter('gravity', 'south_west')).toBe('g_south_west')
      expect(compileParameter('gravity', 'custom:face')).toBe('g_custom:face')
    })
    it('accepts `auto`', () => {
      expect(compileParameter('gravity', 'auto')).toBe('g_auto')
    })
    it('accepts a string starting with `auto:`', () => {
      expect(compileParameter('gravity', 'auto:50')).toBe('g_auto:50')
    })
    it('throws when invalid', () => {
      expect(() => compileParameter('gravity', 'bad:auto')).toThrowErrorMatchingSnapshot()
      expect(() => compileParameter('gravity', 'auto-bad')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('zoom', () => {
    it('accepts a number', () => {
      expect(compileParameter('zoom', 30)).toBe('z_30')
    })
    it('accepts a numeric string', () => {
      expect(compileParameter('zoom', '30')).toBe('z_30')
    })
    it('throws when invalid', () => {
      expect(() => compileParameter('zoom', 'bad')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('x', () => {
    it('accepts a number', () => {
      expect(compileParameter('x', 100)).toBe('x_100')
    })
    it('accepts a numeric string', () => {
      expect(compileParameter('x', '100')).toBe('x_100')
    })
    it('throws when invalid', () => {
      expect(() => compileParameter('x', 'bad')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('y', () => {
    it('accepts a number', () => {
      expect(compileParameter('y', 100)).toBe('y_100')
    })
    it('accepts a numeric string', () => {
      expect(compileParameter('y', '100')).toBe('y_100')
    })
    it('throws when invalid', () => {
      expect(() => compileParameter('y', 'bad')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('fetchFormat', () => {
    it('accepts valid file formats', () => {
      expect(compileParameter('fetchFormat', 'jpg')).toBe('f_jpg')
      expect(compileParameter('fetchFormat', 'png')).toBe('f_png')
    })
    it('accepts `auto`', () => {
      expect(compileParameter('fetchFormat', 'auto')).toBe('f_auto')
    })
    it('throws when invalid', () => {
      expect(() => formatParameter('bad')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('quality', () => {
    it('accepts a number between 1 an 100 inclusive', () => {
      expect(compileParameter('quality', 1)).toBe('q_1')
      expect(compileParameter('quality', 60)).toBe('q_60')
      expect(compileParameter('quality', 100)).toBe('q_100')
    })
    it('accepts a numeric string', () => {
      expect(compileParameter('quality', '50')).toBe('q_50')
    })
    it('accepts a percentage and chroma sub-sampling', () => {
      expect(compileParameter('quality', '60:420')).toBe('q_60:420')
    })
    it('accepts `auto`, auto variants and `jpegmini`', () => {
      expect(compileParameter('quality', 'auto')).toBe('q_auto')
      expect(compileParameter('quality', 'auto:best')).toBe('q_auto:best')
      expect(compileParameter('quality', 'jpegmini')).toBe('q_jpegmini')
    })
    it('throws when invalid', () => {
      expect(() => compileParameter('quality', 0)).toThrowErrorMatchingSnapshot()
      expect(() => compileParameter('quality', 105)).toThrowErrorMatchingSnapshot()
      expect(() => compileParameter('quality', '105')).toThrowErrorMatchingSnapshot()
      expect(() => compileParameter('quality', 'bad')).toThrowErrorMatchingSnapshot()
      expect(() => compileParameter('quality', 'auto:bad')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('radius', () => {
    it('accepts a number', () => {
      expect(compileParameter('radius', 30)).toBe('r_30')
    })
    it('accepts a numeric string', () => {
      expect(compileParameter('radius', '30')).toBe('r_30')
    })
    it('accepts a string of the form x[:y[:z[:u]]]', () => {
      expect(compileParameter('radius', '20:0')).toBe('r_20:0')
      expect(compileParameter('radius', '20:0:40')).toBe('r_20:0:40')
      expect(compileParameter('radius', '20:0:40:40')).toBe('r_20:0:40:40')
    })
    it('throws when invalid', () => {
      expect(() => compileParameter('radius', 'bad')).toThrowErrorMatchingSnapshot()
      expect(() => compileParameter('radius', '20:0:40:40:90')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('angle', () => {
    it('accepts a number', () => {
      expect(compileParameter('angle', 90)).toBe('a_90')
      expect(compileParameter('angle', -20)).toBe('a_-20')
    })
    it('accepts a numeric string', () => {
      expect(compileParameter('angle', '10')).toBe('a_10')
      expect(compileParameter('angle', '-20')).toBe('a_-20')
    })
    it('accepts valid modes', () => {
      expect(compileParameter('angle', 'auto_right')).toBe('a_auto_right')
      expect(compileParameter('angle', 'vflip')).toBe('a_vflip')
    })
    it('throws when invalid', () => {
      expect(() => compileParameter('angle', 'bad')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('effect', () => {
    it('accepts any value', () => {
      expect(compileParameter('effect', 'hue:40')).toBe('e_hue:40')
      expect(compileParameter('effect', 'negate')).toBe('e_negate')
    })
  })

  describe('opacity', () => {
    it('accepts a number between 0 and 100', () => {
      expect(compileParameter('opacity', 0)).toBe('o_0')
      expect(compileParameter('opacity', 30)).toBe('o_30')
      expect(compileParameter('opacity', 100)).toBe('o_100')
    })
    it('accepts a numeric string', () => {
      expect(compileParameter('opacity', '0')).toBe('o_0')
      expect(compileParameter('opacity', '30')).toBe('o_30')
      expect(compileParameter('opacity', '100')).toBe('o_100')
    })
    it('throws when invalid', () => {
      expect(() => compileParameter('opacity', -1)).toThrowErrorMatchingSnapshot()
      expect(() => compileParameter('opacity', 101)).toThrowErrorMatchingSnapshot()
      expect(() => compileParameter('opacity', 'bad')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('border', () => {
    it('accepts a fully formed width_style_color string', () => {
      expect(compileParameter('border', '4px_solid_black')).toBe('bo_4px_solid_black')
      expect(compileParameter('border', '4px_solid_#2040fa')).toBe('bo_4px_solid_rgb:2040fa')
      expect(compileParameter('border', '4px_solid_rgb:2040fa')).toBe('bo_4px_solid_rgb:2040fa')
      expect(compileParameter('border', '4px_solid_rgb:2040faf0')).toBe('bo_4px_solid_rgb:2040faf0')
    })
    it('accepts a object with color and optional width', () => {
      expect(compileParameter('border', {color: 'red'})).toBe('bo_1px_solid_red')
      expect(compileParameter('border', {color: '#3020ff'})).toBe('bo_1px_solid_rgb:3020ff')
      expect(compileParameter('border', {color: '#3020ff22'})).toBe('bo_1px_solid_rgb:3020ff22')
      expect(compileParameter('border', {color: 'rgb:3020ff'})).toBe('bo_1px_solid_rgb:3020ff')
      expect(compileParameter('border', {color: 'rgb:3020ff22'})).toBe('bo_1px_solid_rgb:3020ff22')
      expect(compileParameter('border', {color: 'red', width: 3})).toBe('bo_3px_solid_red')
      expect(compileParameter('border', {color: '#3020ff', width: '2px'})).toBe('bo_2px_solid_rgb:3020ff')
    })
    it('throws when invalid', () => {
      expect(() => compileParameter('border', 'bad')).toThrowErrorMatchingSnapshot()
      expect(() => compileParameter('border', {width: 1})).toThrowErrorMatchingSnapshot()
      expect(() => compileParameter('border', {color: '#3020f'})).toThrowErrorMatchingSnapshot()
      expect(() => compileParameter('border', {color: '#3020ff2'})).toThrowErrorMatchingSnapshot()
    })
  })

  describe('background', () => {
    it('accepts a color', () => {
      expect(compileParameter('background', 'blue')).toBe('b_blue')
      expect(compileParameter('background', '#3020ff')).toBe('b_rgb:3020ff')
      expect(compileParameter('background', '#3020ff22')).toBe('b_rgb:3020ff22')
      expect(compileParameter('background', 'rgb:3020ff')).toBe('b_rgb:3020ff')
      expect(compileParameter('background', 'rgb:3020ff22')).toBe('b_rgb:3020ff22')
    })
    it('accepts valid `auto:mode` strings', () => {
      expect(compileParameter('background', 'auto:border')).toBe('b_auto:border')
      expect(compileParameter('background', 'auto:border_contrast')).toBe('b_auto:border_contrast')
    })
    it('throws when invalid', () => {
      expect(() => compileParameter('background', 78)).toThrowErrorMatchingSnapshot()
      expect(() => compileParameter('background', 'auto:bad')).toThrowErrorMatchingSnapshot()
      expect(() => compileParameter('background', '#3020f')).toThrowErrorMatchingSnapshot()
    })
  })
})

describe('Image Transforms', () => {
  describe('imageTransform', () => {
    it('compiles to a width', () => {
      expect(imageTransform({width: 300})).toBe('w_300')
    })
  })
})
