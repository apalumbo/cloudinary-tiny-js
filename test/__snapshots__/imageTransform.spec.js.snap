// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Transform Parameter angle throws when invalid 1`] = `"Cloudinary Image :: angle should be one of ['auto_right', 'auto_left', 'ignore', 'vflip', 'hflip'] or a number, received: 'bad' - see http://cloudinary.com/documentation/image_transformation_reference#angle_parameter"`;

exports[`Image Transform Parameter aspectRatio throws when invalid 1`] = `"Cloudinary Image :: aspectRatio should be a number or have the form x:y, received: 'bad' - see http://cloudinary.com/documentation/image_transformation_reference#aspectRatio_parameter"`;

exports[`Image Transform Parameter aspectRatio throws when invalid 2`] = `"Cloudinary Image :: aspectRatio should be a number or have the form x:y, received: '7:' - see http://cloudinary.com/documentation/image_transformation_reference#aspectRatio_parameter"`;

exports[`Image Transform Parameter background throws when invalid 1`] = `"Cloudinary Image :: background should be one of ['auto:border', 'auto:predominant', 'auto:border_contrast', 'auto:predominant_contrast'] or a color, received: 78 - see http://cloudinary.com/documentation/image_transformation_reference#background_parameter"`;

exports[`Image Transform Parameter background throws when invalid 2`] = `"Cloudinary Image :: background should be one of ['auto:border', 'auto:predominant', 'auto:border_contrast', 'auto:predominant_contrast'] or a color, received: 'auto:bad' - see http://cloudinary.com/documentation/image_transformation_reference#background_parameter"`;

exports[`Image Transform Parameter background throws when invalid 3`] = `"Cloudinary Image :: background should be one of ['auto:border', 'auto:predominant', 'auto:border_contrast', 'auto:predominant_contrast'] or a color, received: '#3020f' - see http://cloudinary.com/documentation/image_transformation_reference#background_parameter"`;

exports[`Image Transform Parameter border throws when invalid 1`] = `"Cloudinary Image :: border should be an object with 'width' & 'color' or a string of the form 'width_style_color', received: 'bad' - see http://cloudinary.com/documentation/image_transformation_reference#border_parameter"`;

exports[`Image Transform Parameter border throws when invalid 2`] = `"Cloudinary Image :: border should contain a 'color' property when provided as an object, received: '{\\"width\\":1}' - see http://cloudinary.com/documentation/image_transformation_reference#border_parameter"`;

exports[`Image Transform Parameter border throws when invalid 3`] = `"Cloudinary Image :: border should be an object with 'width' & 'color' or a string of the form 'width_style_color', received: '1px_solid_#3020f' - see http://cloudinary.com/documentation/image_transformation_reference#border_parameter"`;

exports[`Image Transform Parameter border throws when invalid 4`] = `"Cloudinary Image :: border should be an object with 'width' & 'color' or a string of the form 'width_style_color', received: '1px_solid_#3020ff2' - see http://cloudinary.com/documentation/image_transformation_reference#border_parameter"`;

exports[`Image Transform Parameter crop throws when invalid 1`] = `"Cloudinary Image :: crop should be one of ['scale', 'fit', 'limit', 'mfit', 'fill', 'lfill', 'pad', 'mpad', 'crop', 'thumb', 'imagga_crop', 'imagga_scale'], received: 'bad' - see http://cloudinary.com/documentation/image_transformation_reference#crop_parameter"`;

exports[`Image Transform Parameter crop throws when invalid 2`] = `"Cloudinary Image :: crop should be one of ['scale', 'fit', 'limit', 'mfit', 'fill', 'lfill', 'pad', 'mpad', 'crop', 'thumb', 'imagga_crop', 'imagga_scale'], received: 300 - see http://cloudinary.com/documentation/image_transformation_reference#crop_parameter"`;

exports[`Image Transform Parameter fetchFormat throws when invalid 1`] = `"Cloudinary Image :: format should be one of ['jpg', 'jpe', 'jpeg', 'jpc', 'jp2', 'j2k', 'wdp', 'jxr', 'hdp', 'png', 'gif', 'webp', 'bmp', 'tif', 'tiff', 'ico', 'pdf', 'ps', 'ept', 'eps', 'eps3', 'psd', 'svg', 'ai', 'djvu', 'flif'], received: 'bad' - see http://cloudinary.com/documentation/image_transformation_reference#format_parameter"`;

exports[`Image Transform Parameter format throws on an invalid extension 1`] = `"Cloudinary Image :: format should be one of ['jpg', 'jpe', 'jpeg', 'jpc', 'jp2', 'j2k', 'wdp', 'jxr', 'hdp', 'png', 'gif', 'webp', 'bmp', 'tif', 'tiff', 'ico', 'pdf', 'ps', 'ept', 'eps', 'eps3', 'psd', 'svg', 'ai', 'djvu', 'flif'], received: 'bad' - see http://cloudinary.com/documentation/image_transformation_reference#format_parameter"`;

exports[`Image Transform Parameter gravity throws when invalid 1`] = `"Cloudinary Image :: gravity should be one of ['north_west', 'north', 'north_east', 'west', 'center', 'east', 'south_west', 'south', 'south_east', 'xy_center', 'face', 'face:center', 'face:auto', 'faces', 'faces:center', 'faces:auto', 'body', 'body:face', 'ocr_text', 'adv_face', 'adv_faces', 'adv_eyes', 'custom', 'custom:face', 'custom:faces', 'custom:adv_face', 'custom:adv_faces'], 'auto', or a string starting with 'auto:', received: 'bad:auto' - see http://cloudinary.com/documentation/image_transformation_reference#gravity_parameter"`;

exports[`Image Transform Parameter gravity throws when invalid 2`] = `"Cloudinary Image :: gravity should be one of ['north_west', 'north', 'north_east', 'west', 'center', 'east', 'south_west', 'south', 'south_east', 'xy_center', 'face', 'face:center', 'face:auto', 'faces', 'faces:center', 'faces:auto', 'body', 'body:face', 'ocr_text', 'adv_face', 'adv_faces', 'adv_eyes', 'custom', 'custom:face', 'custom:faces', 'custom:adv_face', 'custom:adv_faces'], 'auto', or a string starting with 'auto:', received: 'auto-bad' - see http://cloudinary.com/documentation/image_transformation_reference#gravity_parameter"`;

exports[`Image Transform Parameter height throws when invalid 1`] = `"Cloudinary Image :: height should be a number, received: 'bad' - see http://cloudinary.com/documentation/image_transformation_reference#height_parameter"`;

exports[`Image Transform Parameter opacity throws when invalid 1`] = `"Cloudinary Image :: opacity should be a number between 0 and 100, received: -1 - see http://cloudinary.com/documentation/image_transformation_reference#opacity_parameter"`;

exports[`Image Transform Parameter opacity throws when invalid 2`] = `"Cloudinary Image :: opacity should be a number between 0 and 100, received: 101 - see http://cloudinary.com/documentation/image_transformation_reference#opacity_parameter"`;

exports[`Image Transform Parameter opacity throws when invalid 3`] = `"Cloudinary Image :: opacity should be a number between 0 and 100, received: 'bad' - see http://cloudinary.com/documentation/image_transformation_reference#opacity_parameter"`;

exports[`Image Transform Parameter quality throws when invalid 1`] = `"Cloudinary Image :: quality should be one of ['jpg', 'jpe', 'jpeg', 'jpc', 'jp2', 'j2k', 'wdp', 'jxr', 'hdp', 'png', 'gif', 'webp', 'bmp', 'tif', 'tiff', 'ico', 'pdf', 'ps', 'ept', 'eps', 'eps3', 'psd', 'svg', 'ai', 'djvu', 'flif'], a number between 1 and 100, or have the form x:y, received: 0 - see http://cloudinary.com/documentation/image_transformation_reference#quality_parameter"`;

exports[`Image Transform Parameter quality throws when invalid 2`] = `"Cloudinary Image :: quality should be one of ['jpg', 'jpe', 'jpeg', 'jpc', 'jp2', 'j2k', 'wdp', 'jxr', 'hdp', 'png', 'gif', 'webp', 'bmp', 'tif', 'tiff', 'ico', 'pdf', 'ps', 'ept', 'eps', 'eps3', 'psd', 'svg', 'ai', 'djvu', 'flif'], a number between 1 and 100, or have the form x:y, received: 105 - see http://cloudinary.com/documentation/image_transformation_reference#quality_parameter"`;

exports[`Image Transform Parameter quality throws when invalid 3`] = `"Cloudinary Image :: quality should be one of ['jpg', 'jpe', 'jpeg', 'jpc', 'jp2', 'j2k', 'wdp', 'jxr', 'hdp', 'png', 'gif', 'webp', 'bmp', 'tif', 'tiff', 'ico', 'pdf', 'ps', 'ept', 'eps', 'eps3', 'psd', 'svg', 'ai', 'djvu', 'flif'], a number between 1 and 100, or have the form x:y, received: '105' - see http://cloudinary.com/documentation/image_transformation_reference#quality_parameter"`;

exports[`Image Transform Parameter quality throws when invalid 4`] = `"Cloudinary Image :: quality should be one of ['jpg', 'jpe', 'jpeg', 'jpc', 'jp2', 'j2k', 'wdp', 'jxr', 'hdp', 'png', 'gif', 'webp', 'bmp', 'tif', 'tiff', 'ico', 'pdf', 'ps', 'ept', 'eps', 'eps3', 'psd', 'svg', 'ai', 'djvu', 'flif'], a number between 1 and 100, or have the form x:y, received: 'bad' - see http://cloudinary.com/documentation/image_transformation_reference#quality_parameter"`;

exports[`Image Transform Parameter quality throws when invalid 5`] = `"Cloudinary Image :: quality should be one of ['jpg', 'jpe', 'jpeg', 'jpc', 'jp2', 'j2k', 'wdp', 'jxr', 'hdp', 'png', 'gif', 'webp', 'bmp', 'tif', 'tiff', 'ico', 'pdf', 'ps', 'ept', 'eps', 'eps3', 'psd', 'svg', 'ai', 'djvu', 'flif'], a number between 1 and 100, or have the form x:y, received: 'auto:bad' - see http://cloudinary.com/documentation/image_transformation_reference#quality_parameter"`;

exports[`Image Transform Parameter radius throws when invalid 1`] = `"Cloudinary Image :: radius should be a number, 'max' or have the form x[:y[:z[:u]]], received: 'bad' - see http://cloudinary.com/documentation/image_transformation_reference#radius_parameter"`;

exports[`Image Transform Parameter radius throws when invalid 2`] = `"Cloudinary Image :: radius should be a number, 'max' or have the form x[:y[:z[:u]]], received: '20:0:40:40:90' - see http://cloudinary.com/documentation/image_transformation_reference#radius_parameter"`;

exports[`Image Transform Parameter width throws when invalid 1`] = `"Cloudinary Image :: width should be a number, 'auto', or a string starting with 'auto:', received: 'bad:auto' - see http://cloudinary.com/documentation/image_transformation_reference#width_parameter"`;

exports[`Image Transform Parameter width throws when invalid 2`] = `"Cloudinary Image :: width should be a number, 'auto', or a string starting with 'auto:', received: 'auto-bad' - see http://cloudinary.com/documentation/image_transformation_reference#width_parameter"`;

exports[`Image Transform Parameter x throws when invalid 1`] = `"Cloudinary Image :: x should be a number, received: 'bad' - see http://cloudinary.com/documentation/image_transformation_reference#x_parameter"`;

exports[`Image Transform Parameter y throws when invalid 1`] = `"Cloudinary Image :: y should be a number, received: 'bad' - see http://cloudinary.com/documentation/image_transformation_reference#y_parameter"`;

exports[`Image Transform Parameter zoom throws when invalid 1`] = `"Cloudinary Image :: zoom should be a number, received: 'bad' - see http://cloudinary.com/documentation/image_transformation_reference#zoom_parameter"`;
